### ISSUE TITLE
gradient data type error

### ISSUE THREAD

**Issue by ryusangwon** [WeiChengTseng/PCGrad#18]
Hi, I'm trying to use PCGrad, but I'm having trouble with this gradient different type error. This error occurs in line number 80 in pcgrad.py. 

<img width="355" alt="Screenshot 2024-01-10 at 10 17 47 PM" src="https://github.com/WeiChengTseng/Pytorch-PCGrad/assets/78716763/9429e10d-c397-479d-8366-691c6dd70519">

If I make grads[idx].to(torch.bfloat16) to match the data type, the same error occurs after the 259th gradient.

<img width="352" alt="Screenshot 2024-01-10 at 10 17 01 PM" src="https://github.com/WeiChengTseng/Pytorch-PCGrad/assets/78716763/83109259-bd62-46c8-b17c-553f35f80fc1">

Is anyone having this problem? or know how to solve this problem?

**Comment by ZZX-11** [comment]
Do you solve this problem？

**Comment by ryusangwon** [comment]
> Do you solve this problem？

yes I solve this problem by reducing the learning rate.
